### PR TITLE
Homework04

### PR DESCRIPTION
**Pull request by u2107680007911543774e7r**
generics, bank, homework04 branch

### CODE REVIEW

File: src/main/java/homework04/Currency.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package homework04;
+
+import java.math.BigDecimal;
+
+public class Currency extends AssetAbstClass {
+
+	public Currency(BigDecimal balance) {
```
Comment (SerhiiPasko) [inline src/main/java/homework04/Currency.java:7]: It this case this is method don't need - you don't override anything.

File: src/main/java/homework04/Gold.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package homework04;
+
+import java.math.BigDecimal;
+
+public class Gold extends AssetAbstClass {
+
+	public Gold(BigDecimal balance) {
```
Comment (SerhiiPasko) [inline src/main/java/homework04/Gold.java:7]: Also as previous

File: src/main/java/homework04/Main.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		BigDecimal bd3 = new BigDecimal(300.0);
+		Gold gold = new Gold(bd1);
+		Currency cur = new Currency(bd2);
+		SecurityPaper sp = new SecurityPaper(bd3);
+
+		Bank<Gold> bankGold = new Bank<Gold>(1, "First", "Ukraine");
+		Bank<Currency> bankCur = new Bank<Currency>(2, "Second", "USA");
+		Bank<SecurityPaper> bankSP = new Bank<SecurityPaper>(3, "Third", "Australia");
+		bankGold.setAsset(gold);
+		System.out.println(bankGold.toString());
```
Comment (SerhiiPasko) [inline src/main/java/homework04/Main.java:19]: You many time repeat same code. Doing this in Collections (for example, List) or in separate method

File: src/main/java/homework04/SecurityPaper.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package homework04;
+
+import java.math.BigDecimal;
+
+public class SecurityPaper extends AssetAbstClass {
+
+	public SecurityPaper(BigDecimal balance) {
```
Comment (SerhiiPasko) [inline src/main/java/homework04/SecurityPaper.java:7]: Also same to previous

**Review by SerhiiPasko — APPROVED** [review]
Good

**Review by DadizhaO — APPROVED** [review]
ok

**Pull request merged**